samples/sample_multi_transcode/src/pipeline_transcode.cpp

```diff
@@ -4599,25 +4607,22 @@ mfxStatus SafetySurfaceBuffer::ReleaseSurface(mfxFrameSurface1* pSurf)
             if (0 == it->Locked)
             {
                 m_SList.erase(it);
-                m_mutex.Unlock();
+                lock.unlock();
 
                 // event operation should be out of synced context
```

Comment:
Why is it so?

Reply:
Hard to say. I believe it is related to event synchronization somehow. I plan to get rid of ```MSDKEvent``` in favor of ```std::condition_variable``` then I believe this code will be gone

Reply:
I believe that was done to avoid deadlock - because there's internal mutex in Event class